pkg/util/operatorcondition_test.go

```diff
@@ -0,0 +1,66 @@
+package util
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("OperatorCondition", func() {
+	DescribeTable("should return no error when setting the condition, in not-supported environments", func(ci ClusterInfo) {
+		oc, err := NewOperatorCondition(ci, nil, UpgradeableCondition)
+		Expect(err).To(BeNil())
+
+		ctx := context.Background()
+		err = oc.Set(ctx, metav1.ConditionTrue, "Reason", "message")
+		Expect(err).To(BeNil())
+	},
+		Entry("should no-op when not managed by OLM", &ClusterInfoImp{
+			managedByOLM:   false,
+			runningLocally: false,
+		}),
+		Entry("should no-op when running locally", &ClusterInfoImp{
+			managedByOLM:   true,
+			runningLocally: true,
+		}),
+		Entry("should no-op when running locally and not managed by OLM", &ClusterInfoImp{
+			managedByOLM:   false,
+			runningLocally: true,
+		}),
+	)
+
+	// Can't test real operator condition, as the library points to a constant path that is not exists in test env
```

Comment:
if it's not going to work I think we should drop this

Reply:
I hope it will. There is an issue open for that: https://github.com/operator-framework/operator-lib/issues/50